### ISSUE TITLE
Tidy up script/CLI for checking file

### ISSUE THREAD

**Issue by rchan26** [alan-turing-institute/prompto#1]
`run_checks` CLI which runs the script [scripts/run_checks.py](https://github.com/alan-turing-institute/batch-llm/blob/main/src/batch_llm/scripts/run_checks.py) is a bit messy (ported from previous code). I think each model should also have a `checker` method which checks if a line is correct (not a full file as a file can have many models in practice). We should call that `model.checker` method when a model is present.

In other words, all the checks for a particular model should be associated to that model.

We use the script to check a file and potentially move it to the input folder. This file can be anywhere.
